app/assets/javascripts/application.js

```diff
@@ -3,8 +3,8 @@
 //= require jquery.timers
 //= require jquery.cookie
 //= require jquery.throttle-debounce
-//= require bootstrap.tooltip
-//= require bootstrap.dropdown
+//= require popper
+//= require bootstrap-sprockets
```

Comment:
What's the story here? I assume `bootstrap-sprockets` is loading the bootstrap javascript from the gem but what's `popper` and where is that coming from?

Reply:
I'm just following the [README](https://github.com/twbs/bootstrap-rubygem/blob/ef343b781ca1316ba3953295222578a65bd576c0/README.md#a-ruby-on-rails):

> Bootstrap tooltips and popovers depend on popper.js for positioning. The bootstrap gem already depends on the popper_js gem.

> Add Bootstrap dependencies and Bootstrap to your application.js:

> `//= require jquery3`
> `//= require popper`
> `//= require bootstrap-sprockets`

> While bootstrap-sprockets provides individual Bootstrap components for ease of debugging, you may alternatively require the concatenated bootstrap for faster compilation:

> `//= require jquery3`
> `//= require popper`
> `//= require bootstrap`